zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -55,84 +66,196 @@ private ZeebeRocksDbFactory(
 
   @Override
   public ZeebeTransactionDb<ColumnFamilyType> createDb(final File pathName) {
-    return open(
-        pathName,
-        Arrays.stream(columnFamilyTypeClass.getEnumConstants())
-            .map(c -> c.name().toLowerCase().getBytes())
-            .collect(Collectors.toList()));
-  }
-
-  private ZeebeTransactionDb<ColumnFamilyType> open(
-      final File dbDirectory, final List<byte[]> columnFamilyNames) {
-
     final ZeebeTransactionDb<ColumnFamilyType> db;
+    final List<AutoCloseable> closeables = new ArrayList<>();
     try {
-      final List<AutoCloseable> closeables = new ArrayList<>();
-
       // column family options have to be closed as last
-      final ColumnFamilyOptions columnFamilyOptions = createColumnFamilyOptions();
+      final var columnFamilyOptions = createColumnFamilyOptions(closeables);
       closeables.add(columnFamilyOptions);
-
-      final List<ColumnFamilyDescriptor> columnFamilyDescriptors =
-          createFamilyDescriptors(columnFamilyNames, columnFamilyOptions);
-      final DBOptions dbOptions =
-          new DBOptions()
-              .setCreateMissingColumnFamilies(true)
-              .setErrorIfExists(false)
-              .setCreateIfMissing(true)
-              .setParanoidChecks(true);
+      final var dbOptions = createDefaultDbOptions(closeables);
       closeables.add(dbOptions);
 
-      db =
-          ZeebeTransactionDb.openTransactionalDb(
-              dbOptions,
-              dbDirectory.getAbsolutePath(),
-              columnFamilyDescriptors,
-              closeables,
-              columnFamilyTypeClass);
+      final var options = new Options(dbOptions, columnFamilyOptions);
+      closeables.add(options);
+
+      db = ZeebeTransactionDb.openTransactionalDb(options, pathName.getAbsolutePath(), closeables);
 
     } catch (final RocksDBException e) {
-      throw new RuntimeException("Unexpected error occurred trying to open the database", e);
+      CloseHelper.quietCloseAll(closeables);
+      throw new IllegalStateException("Unexpected error occurred trying to open the database", e);
     }
     return db;
   }
 
-  private List<ColumnFamilyDescriptor> createFamilyDescriptors(
-      final List<byte[]> columnFamilyNames, final ColumnFamilyOptions columnFamilyOptions) {
-    final List<ColumnFamilyDescriptor> columnFamilyDescriptors = new ArrayList<>();
+  private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
+    final var statistics = new Statistics();
+    closeables.add(statistics);
+    statistics.setStatsLevel(StatsLevel.ALL);
 
-    if (columnFamilyNames != null && !columnFamilyNames.isEmpty()) {
-      for (final byte[] name : columnFamilyNames) {
-        final ColumnFamilyDescriptor columnFamilyDescriptor =
-            new ColumnFamilyDescriptor(name, columnFamilyOptions);
-        columnFamilyDescriptors.add(columnFamilyDescriptor);
-      }
-    }
-    return columnFamilyDescriptors;
+    return new DBOptions()
```

Comment:
Yes there's an overhead of keeping track of all SST files which grows as you get more files and become noticeable with thousands or more, I removed it in the prototype because I saw we had so few files on both low and high load, and never saw much impact. We could still preemptively set it just in case - RocksDB docs mention this has a performance impact however, as every access to an SST file needs to look up the `fd` in a table (even if it's already open), but it's unclear if it really is for us.

Reply:
https://github.com/zeebe-io/zeebe/issues/6213